### ISSUE TITLE
Refactoring of FieldSet field store

### ISSUE THREAD

**Issue by VeckoTheGecko** [OceanParcels/Parcels#1663]
https://github.com/OceanParcels/Parcels/blob/595b7a94cdf97182df53c88fa99cf05646f07346/parcels/fieldset.py#L162-L197

I don't like the current implementation of `.add_field()` calling `set_attr(self, name, field)`. Programmatically storing these fields messes with static typecheckers, and it would be clearer if this was handled another way (either a list of fields, a dictionary, or something else). We can set [`__getattr__`](https://stackoverflow.com/questions/4295678/understanding-the-difference-between-getattr-and-getattribute) so users can still do `fieldset.U`

**Comment by erikvansebille** [comment]
Agree, this can be made _much_ cleaner. Note though that this could also be touched upon / impacted by the unstructured grids development?
